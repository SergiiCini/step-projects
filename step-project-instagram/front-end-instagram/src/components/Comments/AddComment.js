import React, {useRef} from 'react';
import {makeStyles} from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import {useDispatch} from 'react-redux';
import {addCommentToPost} from '../../redux/posts/postsActions';
import TextField from '@material-ui/core/TextField';

const useStyles = makeStyles((theme) => ({
    form: {
        display: 'flex',
        margin: '0 auto',
        width: '100%',
        '& > *': {
            margin: theme.spacing(1),
        },
    },
    postcomment: {
        width: '10%',
        display: 'flex',

    },
    placeholder: {
        width: '90%',
        display: 'flex',
        resize:'none',        
    },
}));

const AddComment = (props) => {
    const {post, active_user, focus} = props
    const classes = useStyles();
    const dispatch = useDispatch();
    const inputRef = useRef('')
 
    const handleEnterKey = (e)=>{
        if(e.keyCode == '13'){
            handleSubmit(e)
        }
        }
    const handleSubmit = (e) => {
        e.preventDefault();
        dispatch(addCommentToPost(active_user, post._id, {text: inputRef.current.value}))
        inputRef.current.value = ''
        console.log(inputRef.current.value)
    }
        if(focus) {
            inputRef.current.focus()
        }


return (

    <form className={classes.form} noValidate autoComplete="off" onSubmit={handleSubmit} onKeyUp={handleEnterKey} >
        <TextField placeholder="Add comment" inputProps={{'aria-label': 'description'}} className={classes.placeholder}
               inputRef={inputRef} multiline/>
        <Button
            type="submit"
            color="primary"
            className={classes.postcomment}
        >Post</Button>
    </form>
);
}

export default AddComment