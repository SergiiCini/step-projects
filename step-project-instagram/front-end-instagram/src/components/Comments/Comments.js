import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import CardActions from '@material-ui/core/CardActions';
import { NavLink } from 'react-router-dom';
import {toggleModal} from "../../redux/modal/modalActions";
import {useDispatch} from "react-redux";


const useStyles = makeStyles((theme) => ({
  comment: {
    display:'flex',
    // margin:'0 auto',
    width:'100%',
    '& > *': {
    margin: theme.spacing(1),
    },
  },
    text: {
      fontSize: '12px',
      width:'70%',
    },
    username:{
      width:'106px'
    },
    postCommentLink: {
        textDecoration: 'none',
        color: '#013B8A',
        fontWeight: 'bold'
    },
}));


const Comments = (props)=> {
    const {comment,allUsers, modalIsOpen} = props
    const classes = useStyles();
    const dispatch = useDispatch();
    const authorOfCommentObj = allUsers.find( user => user._id === comment.author_id );

  return (

      <CardActions
          className={classes.comment} >
          <NavLink 
              onClick = {()=> {
                  if (modalIsOpen) dispatch(toggleModal())
              }}
              className={classes.postCommentLink}
              to={`/account/${authorOfCommentObj._id}`
              }>
            <Typography className={classes.username}>{authorOfCommentObj.username} </Typography>
          </NavLink>
           <Typography className={classes.text}>{comment.text}</Typography>
      </CardActions>
    
  );
}

export default Comments;