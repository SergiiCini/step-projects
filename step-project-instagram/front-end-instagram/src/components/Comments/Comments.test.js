import React from 'react'
import Comments from './Comments';

describe('Testing comments component',()=>{

    let allUsers 
    let postcomment 
    beforeEach(()=>{
            allUsers = []
            postcomment = {}
    })

    test('comments props',()=>{


        const postcomment = {
            a:'1',
            b:'2'
        }
        const allUsers = [
            {a:'1'},
            {b:'2'}
        ]
        const component = <Comments comment={postcomment} allUsers={allUsers}/>
        expect(component).toMatchSnapshot()
        expect(allUsers).toEqual(expect.any(Array))
        expect(postcomment).toEqual(expect.any(Object))
    })
})
