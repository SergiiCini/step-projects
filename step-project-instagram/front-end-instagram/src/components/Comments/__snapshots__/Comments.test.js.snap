// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing comments component comments props 1`] = `
<Comments
  allUsers={
    Array [
      Object {
        "a": "1",
      },
      Object {
        "b": "2",
      },
    ]
  }
  comment={
    Object {
      "a": "1",
      "b": "2",
    }
  }
/>
`;
